src/test/rpc/GetCounts_test.cpp

```diff
@@ -0,0 +1,119 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012-2017 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <test/jtx.h>
+#include <ripple/beast/unit_test.h>
+#include <ripple/protocol/JsonFields.h>
+#include <ripple/protocol/SField.h>
+
+namespace ripple {
+
+class GetCounts_test : public beast::unit_test::suite
+{
+    void testGetCounts()
+    {
+        using namespace test::jtx;
+
+        Env env(*this);
+        // check counts with no transactions posted
+        auto jrr = env.rpc("get_counts")[jss::result];
+        BEAST_EXPECT(jrr[jss::status] == "success");
+        BEAST_EXPECT(! jrr.isMember("Transaction"));
+        BEAST_EXPECT(! jrr.isMember("STObject"));
+        BEAST_EXPECT(! jrr.isMember("HashRouterEntry"));
+        BEAST_EXPECT(
+            jrr.isMember(jss::uptime) &&
+            ! jrr[jss::uptime].asString().empty());
+        BEAST_EXPECT(
+            jrr.isMember(jss::dbKBTotal) &&
+            jrr[jss::dbKBTotal].asInt() > 0);
+
+        // create some transactions
+        env.close();
+        Account alice {"alice"};
+        Account bob {"bob"};
+        env.fund (XRP(10000), alice, bob);
+        env.trust (alice["USD"](1000), bob);
+        for(auto i=0; i<20; ++i)
+        {
+            env (pay (alice, bob, alice["USD"](5)));
+            env.close();
+        }
+
+        jrr = env.rpc("get_counts")[jss::result];
+        BEAST_EXPECT(jrr[jss::status] == "success");
+        BEAST_EXPECT(
+            jrr.isMember("Transaction") &&
+            jrr["Transaction"].asInt() > 0);
```

Comment:
Hmmm.  I'm a little surprised that you don't think the `CountedObjects` are predicable enough to give reliable results.  But I've never looked at them very hard.  I certainly accept your assessment.

Interesting thought about comparing to the `CountedObject` values.  It would not be worth heroic efforts to have an exact value to compare against.  But if it's relatively easy to do it might be nice.  Your call.